### PR TITLE
MULTIARCH-4358: Agent-based workflow for remote libvirt installs on IBM Power and IBM Z

### PR DESCRIPTION
**Pull request by jaypoulz**
The PR extends the existing deployment options for remote hosts running a libvirt daemon to install using an agent-based workflow. This will augment testing for the IBM Power and IBM Z architectures, and harden our multi-arch validation of the agent-based installer and the assisted service.

### CODE REVIEW

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x-heterogeneous

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

File: ci-operator/step-registry/upi/install/libvirt/upi-install-libvirt-commands.sh [hunk 1]
```diff
@@ -194,34 +267,74 @@ create_vm () {
...
+      --memory ${DOMAIN_MEMORY} \
+      --vcpus ${DOMAIN_VCPUS} \
+      --network network=${CLUSTER_NAME},mac=${MAC_ADDRESS} \
+      --disk="vol=${POOL_NAME}/${NAME}-volume" \
+      --osinfo ${VIRT_INSTALL_OSINFO} \
+      --graphics=none \
+      --import \
+      --noautoconsole \
+      --disk vol=${POOL_NAME}/${IGNITION_VOLUME},format=raw,readonly=on,serial=ignition,startup_policy=optional
+  fi
```
Comment (tvardema) [inline ci-operator/step-registry/upi/install/libvirt/upi-install-libvirt-commands.sh:322]: Adding the above, including volume clone function calls, means the function is no longer simply "creating the vm" as its name implies... Maybe a more appropriate function name would be `create_node` to reflect its expanded functionality?

File: ci-operator/step-registry/upi/install/libvirt/upi-install-libvirt-commands.sh [hunk 2]
```diff
@@ -295,4 +418,5 @@ done
 
 date "+%F %X" > "${SHARED_DIR}/CLUSTER_INSTALL_END_TIME"
 
-touch /tmp/install-complete
+touch ${INSTALL_DIR}/install-complete
+sleep 20m
```
Comment (tvardema) [inline ci-operator/step-registry/upi/install/libvirt/upi-install-libvirt-commands.sh:422]: Is this sleep for you to debug job runs, or is there additional sleep needed for agent based workflows?

File: ci-operator/step-registry/upi/install/libvirt/upi-install-libvirt-ref.yaml [hunk 3]
```diff
@@ -16,6 +16,9 @@ ref:
   - name: DOMAIN_VCPUS
     documentation: Number of VCPUs to allocate for each guest
     default: '6'
+  - name: INSTALLER_TYPE
+    documentation: The type of installer to use for deployment. Supported values are [agent, assisted, base]
+    default: 'base'
```
Comment (tvardema) [inline ci-operator/step-registry/upi/install/libvirt/upi-install-libvirt-ref.yaml:21]: I think I'd prefer `[agent, assisted, default]` as a `base` install-type sounds like it refers to something that would be done as a minimal install or as a first step before doing something more with the installation.  I think its misleading.  (references in if-blocks above would need updated)

**Review by tvardema — COMMENTED** [review]
Lookin pretty solid!  Mostly nits for comments.  Lets make sure when the time comes to get rehearsals for merge that we also rehearse a default install with this workflow, to make sure it didn't get messed up somehow.

File: ci-operator/step-registry/upi/install/libvirt/upi-install-libvirt-commands.sh (hunk 2 above)
Comment (jaypoulz) [reply ci-operator/step-registry/upi/install/libvirt/upi-install-libvirt-commands.sh:422]: As discussed - just debugging for now. will leave this comment thread so I remember to remove.

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le  periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le

File: ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.16.yaml [hunk 4]
```diff
@@ -398,6 +398,21 @@ tests:
       NODE_TUNING: "true"
       TEST_TYPE: conformance-parallel
     workflow: openshift-e2e-libvirt-upi
+- as: ocp-e2e-ovn-agent-remote-libvirt-s390x
+  cluster: build01
+  cron: 0 4 * * 1-5
+  steps:
+    cluster_profile: libvirt-s390x
+    dependencies:
+      OPENSHIFT_INSTALL_TARGET: release:s390x-latest
```
Comment (tvardema) [inline ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.16.yaml:407]: Just curious, did we not want to use the multi-payload for these agent jobs?

**Review by tvardema — APPROVED** [review]
Looks good!

File: ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.16.yaml (hunk 4 above)
Comment (jaypoulz) [reply ci-operator/config/openshift/multiarch/openshift-multiarch-master__nightly-4.16.yaml:407]: There's a bug with agent-based installs that breaks when using the multi-payload.
I've started investigating to see if it's still an issue.

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-agent-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-multi-z-z periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-z-z periodic-ci-openshift-multiarch-master-nightly-4.17-ocp-e2e-ovn-agent-remote-libvirt-multi-p-p

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-multi-z-z

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.17-ocp-e2e-ovn-agent-remote-libvirt-multi-p-p

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-multi-z-z periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-multi-z-z periodic-ci-openshift-multiarch-master-nightly-4.17-ocp-e2e-ovn-agent-remote-libvirt-multi-p-p

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-multi-z-z periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.17-ocp-e2e-ovn-agent-remote-libvirt-multi-z-z

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-multi-p-p

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-multi-p-p

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-agent-remote-libvirt-s390x

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-s390x  periodic-ci-openshift-multiarch-master-nightly-4.17-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.17-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.17-ocp-e2e-ovn-remote-libvirt-multi-p-p

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.17-ocp-e2e-ovn-agent-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.17-ocp-e2e-ovn-remote-libvirt-ppc64le periodic-ci-openshift-multiarch-master-nightly-4.17-ocp-e2e-ovn-remote-libvirt-multi-p-p

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-multi-z-z periodic-ci-openshift-multiarch-master-nightly-4.17-ocp-e2e-ovn-remote-libvirt-multi-z-z periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-multi-z-z

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehease periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-multi-p-p

File: ci-operator/step-registry/upi/conf/libvirt/network/upi-conf-libvirt-network-ref.yaml [hunk 5]
```diff
@@ -2,6 +2,10 @@ ref:
   as: upi-conf-libvirt-network
   from: libvirt-installer
   commands: upi-conf-libvirt-network-commands.sh
+  env:
+  - name: INSTALLER_TYPE
+    documentation: The type of installer to use for deployment. Supported values are [agent, assisted, default]
```
Comment (tvardema) [inline ci-operator/step-registry/upi/conf/libvirt/network/upi-conf-libvirt-network-ref.yaml:7]: Do we currently support `assisted` in the upi workflows?  We should probably only include the options that are fully supported if it isn't.

File: ci-operator/step-registry/upi/install/libvirt/upi-install-libvirt-ref.yaml [hunk 6]
```diff
@@ -16,12 +16,17 @@ ref:
   - name: DOMAIN_VCPUS
     documentation: Number of VCPUs to allocate for each guest
     default: '6'
+  - name: INSTALLER_TYPE
+    documentation: The type of installer to use for deployment. Supported values are [agent, assisted, default]
```
Comment (tvardema) [inline ci-operator/step-registry/upi/install/libvirt/upi-install-libvirt-ref.yaml:20]: Same comment as the other regarding `assisted` support.

File: ci-operator/step-registry/upi/conf/libvirt/network/upi-conf-libvirt-network-ref.yaml (hunk 5 above)
Comment (jaypoulz) [reply ci-operator/step-registry/upi/conf/libvirt/network/upi-conf-libvirt-network-ref.yaml:7]: We do not support assissted yet - this is a good note. I completely forgot about this line.

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-multi-z-z periodic-ci-openshift-multiarch-master-nightly-4.17-ocp-e2e-ovn-remote-libvirt-multi-z-z periodic-ci-openshift-multiarch-master-nightly-4.18-ocp-e2e-ovn-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.17-ocp-e2e-ovn-remote-libvirt-multi-p-p periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-s390x periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le

File: ci-operator/step-registry/upi/conf/libvirt/network/upi-conf-libvirt-network-commands.sh [hunk 7]
```diff
@@ -39,7 +39,54 @@ CLUSTER_NAME="${LEASED_RESOURCE}-${UNIQUE_HASH}"
 BASE_URL="${CLUSTER_NAME}.${BASE_DOMAIN}"
 
 echo "Creating the libvirt network.xml file..."
-cat >> "${SHARED_DIR}/network.xml" << EOF
+
+# If you're wondering why this version of the network exists, I'm sorry.
```
Comment (tvardema) [inline ci-operator/step-registry/upi/conf/libvirt/network/upi-conf-libvirt-network-commands.sh:43]: Author's footnotes?  Lol, unnecessary.

**Comment by jaypoulz** [comment]
This last push was just tweaks to comments, so marking it as rehearsed.

**Review by bradmwilliams — APPROVED** [review]

**Pull request merged**